generator/generator.go

```diff
@@ -487,7 +487,7 @@ func processEmbedded(t ast.Expr, pr typePrinter, input targetProcessInput) (gene
 		x = v.X
 		hasGenericsParams = true
 
-		genericParam, _, err = processEmbedded(v.Index, pr, input)
+		genericParam, _, err = processEmbedded(v.Index, pr, input, false)
```

Comment:
Let's add a comment here:
/* 
 Don't check if embedded interface's generic params are also interfaces, e.g. given the interface:
type SomeInterface[T any] {
   EmbeddedGenericInterface[T]
}

we won't be checking if T is also an interface
*/